src/rebuild/srv.c

```diff
@@ -1462,6 +1477,26 @@ rebuild_fini_one(void *arg)
 	D_ASSERT(dss_get_module_info()->dmi_xs_id != 0);
 	ds_cont_local_close(rpt->rt_coh_uuid);
 
+	dpc = ds_pool_child_lookup(rpt->rt_pool_uuid);
+	D_ASSERT(dpc != NULL);
+
+	/* Reset rebuild epoch, then reset the aggregation epoch, so
+	 * it can aggregate the rebuild epoch.
+	 */
+	D_ASSERT(rpt->rt_rebuild_fence != 0);
+	if (rpt->rt_rebuild_fence == dpc->spc_rebuild_fence) {
+		dpc->spc_rebuild_fence = 0;
+		dpc->spc_rebuild_end_hlc = crt_hlc_get();
+		D_DEBUG(DB_REBUILD, DF_UUID": Reset aggregation end hlc "
+			DF_U64"\n", DP_UUID(rpt->rt_pool_uuid),
+			dpc->spc_rebuild_end_hlc);
+	} else {
```

Comment:
Although only one rebuild is allowed, but there might be multiple rpts due to asynchronous rpt_fini  so when finish rpt, it needs to compare the fence to make sure no other rebuild is going on.

Reply:
OK, I see. Then seems there is a possible problem, that when there are many rebuild tasks queued, it possibly cause the async rebuild_fini_one() with new rebuild task start ahead, then cause the dpc->spc_rebuild_end_hlc cannot be updated for a long time. Within that time if the aggregate task triggered, cont_child_aggregate() the code of :
change_hlc = max(cont->sc_snapshot_delete_hlc,
			 cont->sc_pool->spc_rebuild_end_hlc);

seems will lose some in-rebuilding status due to old spc_rebuild_end_hlc (not updated), right?
If that's true, maybe can just change the final rebuild_iv_update() to work in EAGER mode?

Reply:
Yes, indeed, I thought about that, but 
1. This is somewhat rare case.
2. Even it happens, spc_rebuild_end_hlc will not be set,  it would only delay this rebuild view deletion in the latter time.  Probably ok for the moment.